pkg/producer/util.go

```diff
@@ -0,0 +1,210 @@
+			}
+			flowType2.DstIP = ie.Value.(net.IP).String()
+		case "sourceTransportPort":
+			flowType2.SrcPort = uint32(ie.Value.(uint16))
+		case "destinationTransportPort":
+			flowType2.DstPort = uint32(ie.Value.(uint16))
+		case "protocolIdentifier":
+			flowType2.Proto = uint32(ie.Value.(uint8))
+		case "packetTotalCount":
+			flowType2.PacketsTotal = ie.Value.(uint64)
+		case "octetTotalCount":
+			flowType2.BytesTotal = ie.Value.(uint64)
+		case "packetDeltaCount":
+			flowType2.PacketsDelta = ie.Value.(uint64)
+		case "octetDeltaCount":
+			flowType2.BytesDelta = ie.Value.(uint64)
+		case "reversePacketTotalCount":
+			flowType2.ReversePacketsTotal = ie.Value.(uint64)
+		case "reverseOctetTotalCount":
+			flowType2.ReverseBytesTotal = ie.Value.(uint64)
+		case "reversePacketDeltaCount":
+			flowType2.ReversePacketsDelta = ie.Value.(uint64)
+		case "reverseOctetDeltaCount":
+			flowType2.ReverseBytesDelta = ie.Value.(uint64)
+		case "sourcePodNamespace":
+			flowType2.SrcPodNamespace = ie.Value.(string)
+		case "sourcePodName":
+			flowType2.SrcPodName = ie.Value.(string)
+		case "sourceNodeName":
+			flowType2.SrcNodeName = ie.Value.(string)
+		case "destinationPodNamespace":
+			flowType2.DstPodNamespace = ie.Value.(string)
+		case "destinationPodName":
+			flowType2.DstPodName = ie.Value.(string)
+		case "destinationNodeName":
+			flowType2.DstNodeName = ie.Value.(string)
+		case "destinationClusterIPv4", "destinationClusterIPv6":
+			if flowType2.DstClusterIP != "" {
+				klog.Warningf("Do not expect destination cluster IP: %v to be filled already", flowType2.DstClusterIP)
+			}
+			flowType2.DstClusterIP = ie.Value.(net.IP).String()
+		case "destinationServicePort":
+			flowType2.DstServicePort = uint32(ie.Value.(uint16))
+		case "destinationServicePortName":
+			flowType2.DstServicePortName = ie.Value.(string)
+		case "ingressNetworkPolicyName":
+			flowType2.IngressPolicyName = ie.Value.(string)
+		case "ingressNetworkPolicyNamespace":
+			flowType2.IngressPolicyNamespace = ie.Value.(string)
+		case "egressNetworkPolicyName":
+			flowType2.EgressPolicyName = ie.Value.(string)
+		case "egressNetworkPolicyNamespace":
+			flowType2.EgressPolicyNamespace = ie.Value.(string)
+		default:
+			klog.Warningf("There is no field with name: %v in flow message (.proto schema)", ie.Element.Name)
+		}
+	}
+	return &protobuf.FlowMessage{
+		FlowType: &protobuf.FlowMessage_Flow2{Flow2: flowType2},
+	}
+}
+
+// convertIPFIXMsgToFlowMsgs converts data records in IPFIX message to flow messages
+// in a given proto schema.
+func convertIPFIXMsgToFlowMsgs(msg *entities.Message, convertRecordFunc convertRecordToFlowType) []*protobuf.FlowMessage {
+	set := msg.GetSet()
+	if set.GetSetType() == entities.Template {
+		return nil
+	}
+
+	flowMsgs := make([]*protobuf.FlowMessage, 0)
+	for _, record := range set.GetRecords() {
+		flowMsg := convertRecordFunc(msg, record)
+		flowMsgs = append(flowMsgs, flowMsg)
+	}
```

Comment:
Trivial comment, since you use make on line 204, the following code could be a little more efficient.

        records := set.GetRecords() 
        flowMsgs := make([]*protobuf.FlowMessage, len(records))
   	for i, record := range records {
		flowMsgs[i] = convertRecordFunc(msg, record)
	}

Reply:
Agreed. Initializing fixed size slice makes it efficient.